flask_restful/reqparse.py

```diff
@@ -236,10 +243,17 @@ def add_argument(self, *args, **kwargs):
         See :class:`Argument`'s constructor for documentation on the
         available options.
         """
+
         if len(args) == 1 and isinstance(args[0], self.argument_class):
             self.args.append(args[0])
         else:
             self.args.append(self.argument_class(*args, **kwargs))
+
+        #Do not know what other argument classes are out there
+        if self.trim and type(self.argument_class) is type(Argument):
```

Comment:
With `trim` enabled globally on a `RequestParser` you will also be trying to call `strip()` on non-string types (int, float, boolean, etc) which will cause an `AttributeError` to be thrown. I don't think it makes sense to have the option to enable trim on all arguments because of this.

Also for future reference, you should check the type of argument class using `isinstance(self.argument_class, Argument)`. Calling `type()` on a `class` will just return `type`:

```python
>>> type(Argument)
<type 'type'>
>>> class Foo:
...     pass
...
>>> type(Foo)
<type 'type'>
```

So the logic here does not work how you wanted it to.